src/lightning_lite/plugins/precision/deepspeed.py

```diff
@@ -43,7 +46,7 @@ class DeepSpeedPrecision(Precision):
             If unsupported ``precision`` is provided.
     """
 
-    def __init__(self, precision: Union[str, int], amp_type: str, amp_level: Optional[str] = None) -> None:
+    def __init__(self, precision: Literal[16, 32, "bf16"], amp_type: str, amp_level: Optional[str] = None) -> None:
```

Comment:
Same suggestion

Reply:
Regarding this and the other comments, I spent like an hour on this and it's spreading like a cancer. I have to break it out because the mypy is just forcing me to do it everywhere. 
#14857

Reply:
If it's becoming a larger unnecessary refactor, you could also revert back to Union[str, int] and still use the `PrecisionType` magic enums for equalities

Reply:
yes, reverted. using old logic now. can be re-done in https://github.com/Lightning-AI/lightning/pull/14857